sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreSuite.scala

```diff
@@ -1065,97 +1184,114 @@ class RocksDBStateStoreSuite extends StateStoreSuiteBase[RocksDBStateStoreProvid
     assert(exception.getMessage.contains("Found long, expecting union"))
   }
 
-  testWithColumnFamiliesAndEncodingTypes(
-    "rocksdb range scan multiple non-contiguous ordering columns",
-    TestWithBothChangelogCheckpointingEnabledAndDisabled ) { colFamiliesEnabled =>
-    val testSchema: StructType = StructType(
-      Seq(
-        StructField("ordering1", LongType, false),
-        StructField("key2", StringType, false),
-        StructField("ordering2", IntegerType, false),
-        StructField("string2", StringType, false),
-        StructField("ordering3", DoubleType, false)
+  Seq(Seq(0, 1, 2), Seq(0, 2, 1), Seq(2, 1, 0), Seq(2, 0, 1)).foreach { sortIndexes =>
+    testWithColumnFamiliesAndEncodingTypes(
+      s"rocksdb range scan multiple non-contiguous ordering columns " +
+        s"and sortIndexes=${sortIndexes.mkString(",")}",
+      TestWithBothChangelogCheckpointingEnabledAndDisabled) { colFamiliesEnabled =>
+      val testSchema: StructType = StructType(
+        Seq(
+          StructField("ordering1", LongType, false),
+          StructField("key2", StringType, false),
+          StructField("ordering2", IntegerType, false),
+          StructField("string2", StringType, false),
+          StructField("ordering3", DoubleType, false)
+        )
       )
-    )
 
-    val testSchemaProj = UnsafeProjection.create(Array[DataType](
+      val testSchemaProj = UnsafeProjection.create(Array[DataType](
         immutable.ArraySeq.unsafeWrapArray(testSchema.fields.map(_.dataType)): _*))
-    val rangeScanOrdinals = Seq(0, 2, 4)
-
-    tryWithProviderResource(
-      newStoreProvider(
-        testSchema,
-        RangeKeyScanStateEncoderSpec(testSchema, rangeScanOrdinals),
-        colFamiliesEnabled
-      )
-    ) { provider =>
-      val store = provider.getStore(0)
+      // Multiply by 2 to get the actual ordinals in the row
```

Comment:
nit: it does not seem to simplify the logic. (0, 1, 2) * 2 vs (0, 2, 4) does not have a huge difference given that someone needs to know about the schema to understand this.

Reply:
I see how do you use this to simplify; effectively doing projection. Looks OK to me.

Reply:
cool thx